### PR TITLE
Feature : Overall UI design

### PR DESCRIPTION
**Pull request by ich-wtag**
Overview

[Task-8: Complete UI Design](https://trello.com/c/qYNaSctj/6-task-8-overall-ui-design)
This PR includes Completes UI Design. When there is no tasks available, blank image with title **Add a Task** will be visible.

Types of Changes

- [x] When there is no task, **Add One** will be visible. This text will be changed based on the **Filter State** .
- [x] **Filter buttons** is inactive, if there is no task added.
- [x] By clicking **Search Icon** search **Input Field** visibility is toggled.
- [x] By clicking **Create Button**  **Add Task Field** visibility is toggled.
- [x] By clicking **Cancel Icon**, **Task Input Field** will be cleared.
- [x] If **Add Task Field** is visible **Load More** button will be visible after **8 tasks**, If not **Load More** button will be visible after **9 tasks**.
- [x] The **UI** is made responsive.
- [x] Code base in **script.js** is updated to remove the repeated code and make codes more robust.

[Final-UI-design.webm](https://github.com/ich-wtag/Todo-App_Vanilla_Js/assets/153156568/9e33715f-07df-4f47-84dd-11c702bdbbf4)

### CODE REVIEW

**Review by nmi-wtag — COMMENTED** [review]
**UI Review 1**

image

Making this section clickable would be more user-friendly.

**Review by nmi-wtag — COMMENTED** [review]
**UI Review 2**

image

We can hide the `task__blank-image` when input card is open

**Review by nmi-wtag — COMMENTED** [review]
**UI Revie 3**
image

Task card is not aligned with search and filter

File: js/const.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+export const INCOMPLETE = "incomplete";
+export const COMPLETE = "complete";
+export const ALL = "all";
+export const INITAIL_PAGE = 1;
+export const PAGE_LOAD_COUNT = 9;
+export const SUCCESS = "success";
+export const ERROR = "error";
+
+export const DONEICON = `<svg class="task__button-icon" width="24" height="19" viewBox="0 0 24 19" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (hhu-wtag) [inline js/const.js]:
```suggestion
export const DONE_ICON = `<svg class="task__button-icon" width="24" height="19" viewBox="0 0 24 19" fill="none" xmlns="http://www.w3.org/2000/svg">
```

File: js/script.js [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+        default:
+            return tobeFilteredArray;
+    }
+};
+
+const paginationHandler = () => {
+    if (currentPage < totalPage) {
+        currentPage++;
+    } else {
+        currentPage = INITAIL_PAGE;
```
Comment (hhu-wtag) [inline js/script.js]: typo
```suggestion
        currentPage = INITIAL_PAGE;
```

**Review by hhu-wtag — APPROVED** [review]

**Review by nmi-wtag — CHANGES_REQUESTED** [review]
**UI review 4**
[Screencast from 17-1-24 11:41:41 পূর্বাহ্ণ +06.webm](https://github.com/ich-wtag/Todo-App_Vanilla_Js/assets/148762372/c92b1ea9-265b-4a06-903f-276c38dc14ee)

While we are clicking the cancel button on edit state without updating the task, the response message shouldn't show like "updated successfully".  Basically we didn't update anything.

**Pull request merged**